Source/CombatExtended/Utilities/Tracking/ThingsTrackingModel.cs

```diff
@@ -0,0 +1,210 @@
+            public int createdOn;
+
+            public bool IsValid => thing.Spawned;
+            public int Age => GenTicks.TicksGame - createdOn;
+
+            public ThingPositionInfo(Thing thing)
+            {
+                this.thing = thing;
+                this.createdOn = GenTicks.TicksGame;
+            }
+
+            public override int GetHashCode()
+            {
+                return thing.thingIDNumber;
+            }
+
+            public override bool Equals(object obj)
+            {
+                return obj.GetHashCode() == GetHashCode();
+            }
+
+            public override string ToString()
+            {
+                return $"{thing.ToString()}:{thing.Position.ToString()}:{createdOn.ToString()}";
+            }
+
+            public int CompareTo(ThingPositionInfo other)
+            {
+                return thing.Position.x.CompareTo(other.thing.Position.x);
+            }
+
+            public static bool operator >(ThingPositionInfo operand1, ThingPositionInfo operand2)
+            {
+                return operand1.CompareTo(operand2) == 1;
+            }
+
+            public static bool operator <(ThingPositionInfo operand1, ThingPositionInfo operand2)
+            {
+                return operand1.CompareTo(operand2) == -1;
+            }
+
+            public static bool operator >=(ThingPositionInfo operand1, ThingPositionInfo operand2)
+            {
+                return operand1.CompareTo(operand2) >= 0;
+            }
+
+            public static bool operator <=(ThingPositionInfo operand1, ThingPositionInfo operand2)
+            {
+                return operand1.CompareTo(operand2) <= 0;
+            }
+        }
+
+        public readonly ThingDef def;
+        public readonly ThingsTracker parent;
+        public readonly Map map;
+
+        public Dictionary<Thing, int> indexByThing = new Dictionary<Thing, int>();
+
+        private HashSet<Thing> things = new HashSet<Thing>();
+        private ThingPositionInfo[] sortedThings = new ThingPositionInfo[100];
+
+        public ThingsTrackingModel(ThingDef def, Map map, ThingsTracker parent)
+        {
+            this.def = def;
+            this.map = map;
+            this.parent = parent;
+        }
+
+        public void Register(Thing thing)
+        {
+            if (indexByThing.ContainsKey(thing))
+                return;
+            if (things.Count + 1 >= sortedThings.Length)
+            {
+                ThingPositionInfo[] temp = new ThingPositionInfo[sortedThings.Length * 2];
+                for (int i = 0; i < sortedThings.Length; i++)
+                    temp[i] = sortedThings[i];
+                sortedThings = temp;
+            }
+            if (things.Count == 0)
+            {
+                sortedThings[things.Count] = new ThingPositionInfo(thing);
+                indexByThing[thing] = 0;
+                things.Add(thing);
+                return;
+            }
+            else
```

Comment:
`else` is redundant when the last statement of the previous `if` block is a `return`.

Reply:
It's fine if you disagree or think it's more readable, but if you resolve without a comment I can't tell if you did it accidentally or have something in mind ;)

Reply:
> `else` is redundant when the last statement of the previous `if` block is a `return`.

A better solution, given that this is the end of the function, is to remove the return from the `if` block.  It compiles to exactly the same code, but if later someone wants to add debugging stuff or whatever to the bottom of the function, there's no hidden returns (if you use code folding not having returns vanish is *nice*).